src/mgmt_conn.c

```diff
@@ -523,6 +535,7 @@ uzfs_zvol_mgmt_get_handshake_info(zvol_io_hdr_t *in_hdr, const char *name,
 	 */
 	mgmt_ack->zvol_guid = dsl_dataset_phys(
 	    zv->zv_objset->os_dsl_dataset)->ds_guid;
+	strncpy(mgmt_ack->replica_id, zv->zv_replica_id, REPLICA_ID_LEN);
```

Comment:
lets try to read from zfs property and send rather than maintaining in-memory zv_replica_id.. this will avoid lot of code and any race conditions if any

Reply:
Done!